transform/transformers/cora_transformer.py

```diff
@@ -0,0 +1,349 @@
+import enum
+import os.path
+import re
+from collections import OrderedDict
+from io import StringIO
+
+import dateutil.parser
+from jinja2 import Environment, PackageLoader
+
+from transform.settings import SDX_FTP_IMAGE_PATH, SDX_FTP_DATA_PATH, SDX_FTP_RECEIPT_PATH
+from transform.transformers.image_transformer import ImageTransformer
+from transform.transformers.pdf_transformer_style_cora import CoraPdfTransformerStyle
+
+env = Environment(loader=PackageLoader('transform', 'templates'))
+
+
+class CORATransformer:
+    """
+    This class captures our understanding of the agreed format
+    of the UKIS survey.
+
+    """
+
+    _MAP_YN_10 = {
+        'yes': '1',
+        'no': '0',
+    }
+
+    _MAP_YN_01 = {
+        'yes': '0',
+        'no': '1',
+    }
+
+    _MAP_YN_21 = {
+        'yes': '10',
+        'no': '01',
+    }
+
+    _MAP_IMPORTANCE = {
+        'not important': '0001',
+        'low': '0010',
+        'medium': '0100',
+        'high': '1000',
+    }
+
+    _MAP_PROPORTIONS = {
+        'none': '0001',
+        'less than 40%': '0010',
+        '40-90%': '0100',
+        'over 90%': '1000',
+    }
+
+    class _Format(enum.Enum):
+
+        zeroone = re.compile("^[01]{1}$")
+        onetwo = re.compile("^[12]{1}$")
+        twobin = re.compile("^[0-1]{1,2}$")
+        twodigits = re.compile("^[0-9]{1,2}$")
+        threedigits = re.compile("^[0-9]{1,3}$")
+        sixdigits = re.compile("^[0-9]{1,6}$")
+        sevendigits = re.compile("^[0-9]{1,7}$")
+        onehotfour = re.compile("^(1000|0100|0010|0001|0000)$")
+        yesno = re.compile("^yes|no$")
+        yesnodk = re.compile("^yes|no|(don.+t know)$")
+
+    class _Processor:
+
+        @staticmethod
+        def false(q, d):
```

Comment:
What are q and d

Reply:
There's quite a few of these in this code, we thought it might be a good idea to have a separate card for making the code more readable?

Reply:
I agree the code is a little obtuse , but I did not need to change it to get it to work in memory . So I took the view that perhaps we should refactor , and review test coverage , as part of a seperate ticket ?